aip/general/0162.md

```diff
@@ -27,124 +27,108 @@ to the version of an API as a whole.
-not already used elsewhere.
-
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+When referring to specific revision of a resource, APIs must use the following syntax
+`{resource_name}/revisions/{revision_id}`. For example:
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+publishers/123/books/les-miserables/revisions/c7cfa2a8
 
 ### Getting a revision
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+APIs implementing resource revisions in the standard `Get` method ([AIP-131][]):
 
 ```proto
-message GetBookRequest {
-  // The name of the book.
+// Gets the details of a book revision.
+rpc GetBookRevision(GetBookRevisionRequest) returns (BookRevision) {
+  option (google.api.http) = {
+    get: "/v1/{name=publishers/*/books/*/revisions/*}"
+  };
+}
+
+message GetBookRevisionRequest {
+  // The name of the book revision.
   //   Example: publishers/123/books/les-miserables
   //
   // In order to retrieve a previous revision of the book, also provide
   // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
+  //   Example: publishers/123/books/les-miserables/revisions/c7cfa2a8
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
+      type: "library.googleapis.com/BookRevision"
     }];
 }
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+If the user passes a revision ID that does not exist, the API must fail with a `NOT_FOUND` error.
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+The returned resource revision `should` be a different resource type from the parent resource type.
 
 ### Tagging revisions
 
+There are two types of tags: Server defined and user specified.
+
+Service may support `current`, `latest`, `oldest`, etc as preserved keyword tags.
+
+GET /v1/publishers/{publisher}/books/{book}/revisions/{preserved_keyword}
+
+Note that `publishers/{publisher}/books/{book}/revisions/latest` and `publishers/{publisher}/books/{book}` can differ. The latest revision may differ from the current status of the resource.
```

Comment:
Is something special about `latest` as a revision? Above, it's mentioned as a possible reserved keyword, but there's no explanation as to what `latest` should mean.

I feel like we should be adding guidance here.

Reply:
added some explanation for latest.